src/pyodide/internal/builtin_wrappers.js

```diff
@@ -147,7 +147,10 @@ export function patchFetch(origin) {
 
     // we didn't find it in the disk cache, continue with original fetch
     const response = await origFetch(url, options);
-    const arrayBuffer = await response.arrayBuffer();
+
+    const arrayBuffer = await new Response(response.body.pipeThrough(new DecompressionStream("gzip"))).arrayBuffer();
+
```

Comment:
hmm... I always get nervous when I see decompression immediately going into an in-memory structure, especially given the potential number of copy operations involved in constructing the `arrayBuffer()`. Is there any chance that `DiskCache.put(...)` could just receive a teed `ReadbleStream` branch rather than an ArrayBuffer here? .. something like:

```
const [b1, b2] = response.body.pipeThrough(new DecompressionStream('gzip')).tee();
DiskCache.put(fileName, b1);
return new Response(b2);
```

This would likely be a lot faster and use far less memory.

Reply:
Sounds good! We'll also have to make a call on whether to keep the tars stored on disk compressed or decompressed. 

It's also notable that this code only executes in local dev mode, so memory usage is less of a priority.